imagery/i.biomass/testsuite/test_i_biomass.py

```diff
@@ -0,0 +1,308 @@
+    def setUpClass(cls):
+        """Set up input rasters and configure test environment."""
+        cls.use_temp_region()
+        cls.runModule("g.region", n=10, s=0, e=10, w=0, rows=10, cols=10)
+        cls._create_input_rasters()
+
+    @classmethod
+    def _create_input_rasters(cls):
+        """Helper method to create all input rasters with defined expressions."""
+        expressions = {
+            "fpar": "col() * 0.1",
+            "lightuse_eff": "row() * 0.1",
+            "latitude": "45.0",
+            "dayofyear": "150",
+            "transmissivity": "0.75",
+            "water": "0.8",
+        }
+        for raster_key, expr in expressions.items():
+            raster_name = cls.input_rasters[raster_key]
+            cls.runModule(
+                "r.mapcalc", expression=f"{raster_name} = {expr}", overwrite=True
+            )
+
+    @classmethod
+    def _create_reference_raster(cls):
+        """Create the reference raster for regression testing."""
+        cls.runModule(
+            "i.biomass",
+            fpar=cls.input_rasters["fpar"],
+            lightuse_efficiency=cls.input_rasters["lightuse_eff"],
+            latitude=cls.input_rasters["latitude"],
+            dayofyear=cls.input_rasters["dayofyear"],
+            transmissivity_singleway=cls.input_rasters["transmissivity"],
+            water_availability=cls.input_rasters["water"],
+            output="biomass_reference",
+            overwrite=True,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up generated data and reset the region."""
+        rasters_to_remove = list(cls.input_rasters.values()) + [cls.output_raster]
+        cls.runModule(
+            "g.remove",
+            type="raster",
+            name=",".join(rasters_to_remove),
+            flags="f",
+            quiet=True,
+        )
+        cls.del_temp_region()
+
+    def test_biomass_regression(self):
+        """Regression test to ensure i.biomass output remains consistent."""
+        self._create_reference_raster()
+        reference_raster = "biomass_reference"
+
+        self.assertModule(
+            "i.biomass",
+            fpar=self.input_rasters["fpar"],
+            lightuse_efficiency=self.input_rasters["lightuse_eff"],
+            latitude=self.input_rasters["latitude"],
+            dayofyear=self.input_rasters["dayofyear"],
+            transmissivity_singleway=self.input_rasters["transmissivity"],
+            water_availability=self.input_rasters["water"],
+            output=self.output_raster,
+            overwrite=True,
+        )
+        self.assertRasterExists(self.output_raster)
+        self.assertRasterExists(reference_raster)
+
+        output_values = array.array(self.output_raster)
+        reference_values = array.array(reference_raster)
+
+        self.assertTrue(
+            np.allclose(output_values, reference_values, atol=2.0),
+            "Biomass raster values should match the reference values",
+        )
```

Comment:
That's not what I had in mind. You are computing the reference raster with each test run, but the idea is to compute the reference at the moment you are writing the test and then this reference is considered to be correct, the values are fixed in the test. So when somebody later changes the tool, resulting in different results, the test will fail and you will know that there is potentially a newly introduced bug. See for example r.viewshed test. It is often enough to compute univariate statistics with r.univar and then test those statistics, not cell-by-cell comparison.